src/main/java/picard/illumina/BasecallsConverterBuilder.java

```diff
@@ -71,6 +71,17 @@ public BasecallsConverterBuilder(final File basecallsDir, final Integer lane, fi
                 outputRecordClass, bclQualityEvaluationStrategy, ignoreUnexpectedBarcodes, applyEamssFiltering, includeNonPfReads);
     }
 
+    /**
+     * Builds a basecalls converter without sorting
+     *
+     * @return A basecalls converter that will output unsorted records.
+     */
+    public BasecallsConverter<CLUSTER_OUTPUT_RECORD> build() {
```

Comment:
How would you feel about adding a method something like:

````java
public ... withSorting(final Comparator<CLUSTER_OUTPUT_RECORD> outputRecordComparator,
                       final SortingCollection.Codec<CLUSTER_OUTPUT_RECORD> codecPrototype,
                       final Class<CLUSTER_OUTPUT_RECORD> outputRecordClass,
                       final List<File> tmpDirs)
```

and then going back to having a single `build()` method?  I think we could even add to the builder a `withMaxRecordsInRam(int max)`  (with some sensible default if it's not called) so that is no longer in the `build()` signature either.  

I think it would be ideal if the builder took in the maximum as just a total maximum and then was allowed to figure out how to use it.  I.e. for a sorting version it should figure that each thread will have a tile, so set max per tile to `max / threads` or something like that.  And then the non-sorting one can use that to configure how big the buffers should be on the output writers when we move to the async pool?  Or is that done somewhere else?